site/src/pages/TemplateSettingsPage/TemplateGeneralSettingsPage/TemplateSettingsPage.test.tsx

```diff
@@ -126,4 +131,66 @@ describe("TemplateSettingsPage", () => {
     const validate = () => getValidationSchema().validateSync(values);
     expect(validate).toThrowError();
   });
+
+  describe("Deprecate template", () => {
+    it("deprecates a template when has access control", async () => {
+      server.use(
+        http.get("/api/v2/entitlements", () => {
+          return HttpResponse.json({
+            ...MockEntitlements,
+            features: API.withDefaultFeatures({
+              access_control: { enabled: true, entitlement: "entitled" },
+            }),
+          });
+        }),
+      );
+      const updateTemplateMetaSpy = jest.spyOn(API, "updateTemplateMeta");
+      const deprecationMessage = "This template is deprecated";
+
+      await renderTemplateSettingsPage();
+      await deprecateTemplate(MockTemplate, deprecationMessage);
+
+      const [templateId, data] = updateTemplateMetaSpy.mock.calls[0];
+      await waitFor(() => expect(templateId).toEqual(MockTemplate.id));
```

Comment:
Does this `waitFor` do anything?  It looks like it is comparing two static IDs so the result would never change, or am I missing something?

Reply:
You are right